extensions/oplogPopulator/modules/ConnectorsManager.js

```diff
@@ -296,14 +291,59 @@ class ConnectorsManager {
                 });
             }
         });
-        const logMessage = connectorUpdateFailed ? 'Failed to update some or all the connectors' :
-            'Successfully updated all the connectors';
-        const logFunction = connectorUpdateFailed ? this._logger.error.bind(this._logger) :
-            this._logger.info.bind(this._logger);
-        logFunction(logMessage, {
-            method: 'ConnectorsManager._updateConnectors',
-            connectorsStatus,
+        if (Object.keys(connectorsStatus).length > 0) {
+            this._logger.info('Successfully updated connectors', {
+                method: 'ConnectorsManager._updateConnectors',
+                connectorsStatus,
+            });
+        }
+    }
+
+    /**
+     * Checks if connectors are in a failed state and restarts them
+     * @returns {undefined}
+     */
+    async _validateConnectorsState() {
+        const restartedConnectors = [];
+        await eachLimit(this._connectors, 10, async connector => {
+            if (!connector.isRunning) {
+                return;
+            }
+
+            try {
+                const connectorStatus = await this._kafkaConnect.getConnectorStatus(connector.name);
+                const isConnectorFailed = connectorStatus?.connector?.state === 'FAILED';
+                const areTasksFailed = connectorStatus?.tasks?.some(task => {
+                    if (task.state === 'FAILED') {
+                        this._logger.error('Connector task failed', {
+                            method: 'ConnectorsManager._validateConnectorsState',
+                            connector: connector.name,
+                            taskId: task.id,
+                            error: task.trace,
+                        });
+                        return true;
+                    }
+                    return false;
+                });
+                if (isConnectorFailed || areTasksFailed) {
+                    await connector.restart();
+                    this._metricsHandler.onConnectorRestart(connector);
+                    restartedConnectors.push(connector.name);
+                }
+            } catch (err) {
+                this._logger.error('Could not check or reset connector state', {
+                    method: 'ConnectorsManager._validateConnectorsState',
+                    connector: connector.name,
+                    error: err.description || err.message,
+                });
+            }
         });
+        if (restartedConnectors.length > 0) {
+            this._logger.info('Successfully restarted connectors', {
```

Comment:
restart should seldom happen : may be simpler to just log each restart on success (e.g. line 331)